pennylane/templates/subroutines/controlled_sequence.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ControlledSequence template.
+"""
+from copy import copy
+import pennylane as qml
+
+from pennylane.operation import Operation
+from pennylane.wires import Wires
+from pennylane.ops.op_math.symbolicop import SymbolicOp
+
+
+class ControlledSequence(SymbolicOp, Operation):
+    r"""Creates a sequence of controlled gates raised to decreasing powers of 2. Can be used as
+    a sub-block in building a `quantum phase estimation <https://en.wikipedia.org/wiki/Quantum_phase_estimation_algorithm>`__
+    circuit.
+
+    Given an :class:`~.Operator` and a list of control wires, this template creates a sequence of
+    controlled gates, one for each control wire, with the base :class:`~.Operator` raised to
+    decreasing powers of 2:
+
+    .. figure:: ../../_static/templates/subroutines/big_ctrl.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    Args:
+        base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
+        control (Union[Wires, Sequence[int], or int]): the wires to be used for control
+
+    Raises:
+        ValueError: if the wires in ``control`` and wires on the ``base`` operator share a common
+            element
+
+    .. seealso:: :class:`~.QuantumPhaseEstimation`
+
+    **Example**
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires = 4)
+
+        @qml.qnode(dev)
+        def circuit():
+
+            for i in range(3):
+                qml.Hadamard(wires = i)
+
+            qml.ControlledSequence(qml.RX(0.25, wires = 3), control = [0, 1, 2])
+
+            qml.adjoint(qml.QFT)(wires = range(3))
+
+            return qml.probs(wires = range(3))
+
+    >>> print(circuit())
+    [0.92059345 0.02637178 0.00729619 0.00423258 0.00360545 0.00423258 0.00729619 0.02637178]
+
+    """
+
+    def _flatten(self):
+        return (self.base,), (self.control,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        return cls(data[0], control=metadata[0])
+
+    def __init__(self, base, control, id=None):
```

Comment:
If we're allowing more than one control wire, should we also have a "`work_wires`" keyword argument? Otherwise, we probably won't be able to decompose this enough on generic devices.

Reply:
Yes, but this can be added afterwards depending on usage